File: src/main/java/com/openlattice/chronicle/services/download/DataDownloadManager.java
package com.openlattice.chronicle.services.download;

import java.util.Map;
import java.util.Set;
import java.util.UUID;

/**
 * @author alfoncenzioka &lt;alfonce@openlattice.com&gt;
 */
public interface DataDownloadManager {
    ParticipantDataIterable getAllParticipantData(
            UUID organizationId,
            UUID studyId,
            UUID participantEntityId,
            String token );

    ParticipantDataIterable getAllPreprocessedParticipantData(
            UUID organizationId,
            UUID studyId,
            UUID participantEntityId,
            String token );

    ParticipantDataIterable getAllParticipantAppsUsageData(
            UUID organizationId,
            UUID studyId,
            UUID participantEntityId,
            String token );

}

File: src/main/java/com/openlattice/chronicle/services/download/DataDownloadManager.kt
package com.openlattice.chronicle.services.download

import com.openlattice.chronicle.constants.ParticipantDataType
import java.util.*

/**
 * @author alfoncenzioka &lt;alfonce@openlattice.com&gt;
 */
interface DataDownloadManager {

    fun getParticipantData(
            organizationId: UUID?,
            studyId: UUID,
            participantEntityId: UUID,
            dataType: ParticipantDataType,
            token: String?): Iterable<Map<String, Set<Any>>>
}